libraries/botbuilder-dialogs/src/dialogContextError.ts

```diff
@@ -0,0 +1,37 @@
+import { DialogContext } from './dialogContext';
+import { DialogInstance } from './dialog';
+
+/**
+ * An Error that includes extra dialog context, including the dialog stack
+ */
+export class DialogContextError extends Error {
+    public readonly dialogContext: {
+        activeDialog?: string;
+        parent?: string;
+        stack: DialogInstance[];
+    };
+
+    /**
+     * Construct a DialogError
+     * @param error Source error
+     * @param dialogContext Dialog context that is the source of the error
+     */
+    constructor(source: Error | string, dialogContext: DialogContext) {
```

Comment:
Because the SDK targets JS and TS developers and this is a public class, the `source` and `dialogContext` parameters should have falsy checks.